tests/main/views/test_frameworks.py

```diff
@@ -161,6 +167,31 @@ def test_create_fails_if_slug_is_invalid(self):
         assert response.status_code == 400
         assert json.loads(response.get_data(as_text=True))["error"] == "Invalid slug value 'this is/invalid'"
 
+    def test_create_fails_if_direct_award_and_further_competition_false(self):
+        framework = self.framework()
+        framework['frameworks']['hasDirectAward'] = False
+        framework['frameworks']['hasFurtherCompetition'] = False
```

Comment:
Related to George's comment above - could we test these with initial `None` values as well, to simulate the pre-default database state?

Reply:
The `None` case will only exist for an interim period until the follow-up PR can make the column  non-nullable. Given it's a small table and nothing new will be going into it between now and (basically) 12 months from now, I don't think it's so important to test this case. Maybe?

Reply:
Hmm you're right, I guess it's probably not vital for this table.

Reply:
Yep if we're definitely going back and making this non-nullable then that's fine.

But in that case I'd expected to see a data migration that set values for any existing fields? (Wouldn't I? Maybe not - I understand we have a policy of not really doing data migrations, but I don't follow entirely why not having them would be better, especially in cases like this where if the UPDATEs aren't done, the migration to set the field to not-nullable will just fail.)

Anyhoo, happy to approve, my not understanding our migrations policy definitely isn't a blocker :-)

Reply:
Something like this? ;) https://github.com/alphagov/digitalmarketplace-api/pull/785/files#diff-3b681bde687c7d636bfdb6a9d03d6f14